src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -871,6 +871,24 @@ private static void SetPathAndQueryParameters(CodeMethod target, OpenApiUrlTreeN
     }
     private CodeTypeBase CreateComposedModelDeclaration(OpenApiUrlTreeNode currentNode, OpenApiSchema schema, OpenApiOperation operation, string suffixForInlineSchema, CodeNamespace codeNamespace) {
         var typeName = currentNode.GetClassName(config.StructuredMimeTypes, operation: operation, suffix: suffixForInlineSchema, schema: schema).CleanupSymbolName();
+        var typesCount = schema.AnyOf?.Count ?? schema.OneOf?.Count ?? 0;
+        if ((typesCount == 1 && schema.Nullable && schema.IsAnyOf()) || // nullable on the root schema outside of anyOf
+            typesCount == 2 && schema.AnyOf.Any(static x => // nullable on a schema in the anyOf
+                                                        x.Nullable &&
+                                                        !x.Properties.Any() &&
+                                                        !x.IsOneOf() &&
+                                                        !x.IsAnyOf() &&
+                                                        !x.IsAllOf() &&
+                                                        !x.IsArray() &&
+                                                        !x.IsReferencedSchema())) { // once openAPI 3.1 is supported, there will be a third case oneOf with Ref and type null.
+            var targetSchema = schema.AnyOf.First(static x => !string.IsNullOrEmpty(x.GetSchemaName()));
+            var className = targetSchema.GetSchemaName().CleanupSymbolName();
+            var shortestNamespace = GetShortestNamespace(codeNamespace, targetSchema);
+            return new CodeType {
+                TypeDefinition = AddModelDeclarationIfDoesntExist(currentNode, targetSchema, className, shortestNamespace),
+                Name = typeName,
```

Comment:
```suggestion
                Name = className,
```
Otherwise, we have types generated with names from the `OpenApiUrlTreeNode` instead of the `OpenApiSchema` as we did previously

Reply:
good catch! thanks!